eZ/Publish/Core/MVC/Symfony/ExpressionLanguage/TwigVariableProviderExtension.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\Core\MVC\Symfony\ExpressionLanguage;
+
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\MVC\Symfony\View\ContentView;
+use eZ\Publish\Core\MVC\Symfony\View\VariableProviderRegistry;
+use Symfony\Component\ExpressionLanguage\ExpressionFunction;
+use Symfony\Component\ExpressionLanguage\ExpressionFunctionProviderInterface;
+
+final class TwigVariableProviderExtension implements ExpressionFunctionProviderInterface
+{
+    public const PROVIDER_REGISTRY_PARAMETER = 'providerRegistry';
+    public const VIEW_PARAMETER = 'view';
+
+    /**
+     * @return \Symfony\Component\ExpressionLanguage\ExpressionFunction[]
+     */
+    public function getFunctions(): array
+    {
+        return [
+            new ExpressionFunction(
+                'provider',
```

Comment:
Naming: too generic, has to contain brief information that it is twig variable related somehow.